docs/specs/thrid-party-client-draft.md

```diff
@@ -0,0 +1,38 @@
+## Third-party Clients
+
+The purpose of this document is to shape the third-party client feature. After the details become concrete, this document will be removed and merged to [OIDC](./oidc.md).
+
+## Authorization Flow
+
+- Remove `is_first_party` flag in oauth client, all existing OAuth clients should be first-party apps. Instead, adding `third_party_app` as new `x_application_type`.
+- Third-party client doesn't support anonymous user, biometric login and re-authentication.
+- Allow using `"https://authgear.com/scopes/full-access"` scope for the third-party client, which allows requesting full user's info.
+    - Update checking scope in userinfo endpoint?
+- Update authorization endpoint accepts requests without pkce code challenge
```

Comment:
Worth mentioning request without PKCE MUST have client secret. Of course, request with client secret also can have PKCE.